lib/puppet/util/windows/process.rb

```diff
@@ -62,6 +66,29 @@ def get_current_process
   end
   module_function :get_current_process
 
+  def open_process(desired_access, inherit_handle, process_id, &block)
+    phandle = nil
+    begin
+      FFI::MemoryPointer.new(:handle, 1) do |phandle_ptr|
+        phandle_ptr = OpenProcess(desired_access, inherit_handle, process_id)
```

Comment:
Minor style issue - I think we've tended to pass booleans as Ruby booleans, rather than `FFI` scoped values. So that we would end up with something like this: 

```ruby
inherit = inherit_handle ? `FFI::WIN32_TRUE` : `FFI::WIN32_FALSE`
phandle_ptr = OpenProcess(desired_access, inherit, process_id)
```